### PR TITLE
Use modern material theme for docs

### PR DESCRIPTION
**Pull request by lc0rp**
### Background
We're using a ReadTheDocs theme that hasn't changed in years, and isn't the default, more modern MkDocs theme.

@RaghavKumar [suggested](https://discord.com/channels/1092243196446249134/1092243197117337679/1120008138091929710) the Material UI theme, and after using it, the docs feel way more accessible.

It's also the most popular MkDocs theme, by far, according to https://github.com/mkdocs/mkdocs/wiki/MkDocs-Themes.

### Changes
- Updated requirements.txt to include the mkdocs-material package
- Changed theme in mkdocs.yaml
- Configured some theme options like color schemes, navigation, etcetera
- Added logo to docs/imgs (for favicon)

### Documentation
N/A - these are the docs

### Test Plan
Played around with it in-browser for the past 3h :-(

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes. 
- [ ] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```
### Preview:
image

image

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -37,8 +37,31 @@ nav:
...
+        icon: material/weather-night
+        name: Switch to dark mode
 
+    # Palette toggle for dark mode
+    - media: "(prefers-color-scheme: dark)"
+      scheme: slate
+      toggle:
+        icon: material/weather-sunny
+        name: Switch to light mode
+
```
Comment (collijk) [inline mkdocs.yml]:
```suggestion
plugins:
    - search
```

Comment (collijk) [reply mkdocs.yml]: Docs should be searchable :-)

**Comment by collijk** [comment]
I think this is a great idea.  I love material.  Some things from my own technical documentation projects.

In the theme you can add:

```yaml
theme:
    name: material
    features:
      - content.code.copy
```
and that will enable the little click to copy box in code snippets for people looking at our docs.

I also just blanket put in 

```yaml
markdown_extensions:
  - abbr
  - admonition
  - attr_list
  - def_list
  - footnotes
  - md_in_html
  - pymdownx.arithmatex:
      generic: true
  - pymdownx.critic
  - pymdownx.caret
  - pymdownx.details
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - pymdownx.highlight:
      anchor_linenums: true
      line_spans: __span
      pygments_lang_class: true
  - pymdownx.inlinehilite
  - pymdownx.keys
  - pymdownx.mark
  - pymdownx.snippets:
      auto_append:
        - includes/abbreviations.md
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
  - pymdownx.tabbed:
      alternate_style: true
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.tilde
  - tables
plugins:
  - table-reader
  - search
extra_javascript:
  - https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js
  - _javascript/tablesort.js
  - _javascript/mathjax.js
  - https://polyfill.io/v3/polyfill.min.js?features=es6
  - https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
```

and add 

```js
window.MathJax = {
  tex: {
    inlineMath: [["\\(", "\\)"]],
    displayMath: [["\\[", "\\]"]],
    processEscapes: true,
    processEnvironments: true
  },
  options: {
    ignoreHtmlClass: ".*|",
    processHtmlClass: "arithmatex"
  }
};

document$.subscribe(() => {
  MathJax.typesetPromise()
})
```
to docs/_javascript/mathjax.js

and 

```js
document$.subscribe(function() {
  var tables = document.querySelectorAll("article table:not([class])")
  tables.forEach(function(table) {
    new Tablesort(table)
  })
})
```
to docs/_javascript/tablesort.js.

This set of stuff enables all the features in material (which are turned off by default).

**Comment by collijk** [comment]
Happy to let you decide what you want to include here, but would like to hear your thoughts.

**Comment by lc0rp** [comment]
These are great! Testing them now...

**Comment by Pwuts** [comment]
It seems to be missing the extension that does highlighting for indented code blocks
image

**Review by ntindle — DISMISSED** [review]

File: mkdocs.yml [hunk 2]
```diff
@@ -7,39 +7,113 @@ nav:
...
+    # Palette toggle for dark mode
+    - media: "(prefers-color-scheme: dark)"
+      scheme: slate
+      toggle:
+        icon: material/weather-sunny
+        name: Switch to light mode
 
 markdown_extensions:
-  admonition:
-  codehilite:
```
Comment (Pwuts) [inline mkdocs.yml:44]: This is used in indented code blocks and should not be removed without updating those blocks

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 2 above)
Comment (lc0rp) [reply mkdocs.yml:44]: Good catch. I've updated the codeblocks.

codehilite (extension) is superceded by Highlight, but requires changes to the code blocks.
Ref: https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/#highlight

**Comment by vittorio88** [comment]
@lc0rp @collijk I made a small correction to a sample YAML snippet that came from this PR. Would appreciate a review, thank you. 
The PR is #5168 .

**Pull request merged**